org.adempiere.base/src/org/compiere/model/MSequence.java

```diff
@@ -564,6 +589,48 @@ public static String getDocumentNoFromSeq(MSequence seq, String trxName, PO po)
 				+ " - Sequence=" + AD_Sequence_ID + " [" + trx + "]");
 		return documentNo;
 	}
+	
+	private Boolean isSequenceNoLevel = null;
+	private Boolean isUsePrefixAsKey = null;
+	private Boolean isUseSuffixAsKey = null;
+	
+	/**
+	 * Is the sequence a sequence no level sequence
+	 * @param seq sequence
+	 * @return true if the sequence is a sequence no level sequence
+	 */
+	public boolean isSequenceNoLevel() {
+		if (isSequenceNoLevel == null) {
+			isSequenceNoLevel = Boolean.valueOf(
+					isUsePrefixAsKey() 
+					|| isUseSuffixAsKey()
+					|| isStartNewYear() 
+					|| isOrgLevelSequence());
+		}
+		return isSequenceNoLevel.booleanValue();
+	}
+	
+	/**
+	 * Is the prefix used as key for the sequence no
+	 * @return true if the prefix starts with prefix key
+	 */
+	public boolean isUsePrefixAsKey() {
+		if (isUsePrefixAsKey == null) {
+			isUsePrefixAsKey = Boolean.valueOf(!Util.isEmpty(getPrefix()) && getPrefix().contains(KEY_CONTEXT_VARIABLE+"@"));
```

Comment:
I think it is simpler to just use "/K" suffix as Key indicator for the "Prefix" and "Suffix" string.